src/sync/rwlock.rs

```diff
@@ -0,0 +1,140 @@
+use crate::rt;
+
+use std::{
+    ops,
+    sync::{LockResult, TryLockError, TryLockResult},
+};
+
+/// Mock implementatoin of `std::sync::RwLock`
+#[derive(Debug)]
+pub struct RwLock<T> {
+    object: rt::RwLock,
+    data: std::sync::RwLock<T>,
+}
+
+/// Mock implementation of `std::sync::RwLockReadGuard`
+#[derive(Debug)]
+pub struct RwLockReadGuard<'a, T> {
+    lock: &'a RwLock<T>,
+    data: Option<std::sync::RwLockReadGuard<'a, T>>,
```

Comment:
It's not immediately obvious why this is an `Option`. I think it's it so that the `Drop` impl can drop the `std` guard and release the `std` lock _before_ releasing the `loom` mock lock, because that might cause another thread to acquire the lock? 

I think a comment explaining this (either here on the field or in the `Drop` impl) would be very helpful.